app/views/checkout/thankyou.scala.html

```diff
@@ -36,6 +36,47 @@
         currency: '@subscription.currency',
         subscriptionId: '@subscription.id.get'
     };
+
+    function getCookie(name) {
+        var nameEQ = name + '=';
+        var ca = document.cookie.split(';');
+        for (var i = 0; i < ca.length; i++) {
+            var c = ca[i];
+            while (c.charAt(0) === ' ') { c = c.substring(1, c.length); }
+            if (c.indexOf(nameEQ) === 0) { return c.substring(nameEQ.length, c.length); }
+        }
+        return null;
+    }
+
+    function getAwinChannel() {
+        const channel = getCookie('gu_referrer_channel');
+        if (channel === null) {
+            return 'direct'; //No referrer the user came direct to the site
+        }
+
+        const values = channel ? channel.split('&') : null;
+        if (values && values[0] === 'afl' && values[1] === 'awin'){
+            return 'aw';
+        }
+        return 'na'; //A referrer other than Awin
+    }
+
+    //<![CDATA[ /*** Do not change ***/
+    var AWIN = {};
+    AWIN.Tracking = {};
+    AWIN.Tracking.Sale = {};
+
+    /*** Set your transaction parameters ***/
+    const productData = guardian.pageInfo.productData;
+    AWIN.Tracking.Sale.amount = productData.amount;
+    AWIN.Tracking.Sale.orderRef = productData.subscriptionId;
+    AWIN.Tracking.Sale.parts = productData.productSegment + ':' + productData.amount;
+    AWIN.Tracking.Sale.voucher = productData.promoCode;
+    AWIN.Tracking.Sale.currency = productData.currency;
+    AWIN.Tracking.Sale.test = '0';
+    AWIN.Tracking.Sale.channel = getAwinChannel();
+    //]]>
```

Comment:
Is it worth wrapping this in the classic `(function() {})()` and explicitly putting `AWIN` on the `window` as before? That way `getCookie`, `getAwinChannel` and `productData` won't bleed out into the global scope.

Reply:
I want to keep this code exactly as provided for the purposes of this test. I'm planning to refactor after we get it working.